### PR TITLE
Implement Chip

### PR DESCRIPTION
**Pull request by magnus-gross**
Implements the Chip as defined in: https://material.io/guidelines/components/chips.html

alt tag

### CODE REVIEW

File: src/imports/controls/Chip.qml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      The caption to show in the chip
+    */
+    property string caption: ""
+
+    /*!
+      \qmlproperty url iconSource
+
+      The icon to display (if any)
+    */
+    property url iconSource: ""
```
Comment (plfiorini) [inline src/imports/controls/Chip.qml]: this has to be done with a grouped `icon` property and an `Icon` instead of `Image`

**Comment by plfiorini** [comment]
Commit message needs a `Closes: #59`

**Comment by oKcerG** [comment]
The delete button doesn't look like what's in the guideline.

Should we be able to specify an image and it will be displayed with a circle mask?
Should we be able to specify a letter (or two?) or an icon and it will be displayed with a circular colored background?

File: src/imports/controls/Chip.qml (hunk 1 above)
Comment (magnus-gross) [reply src/imports/controls/Chip.qml]: Wouldn't an Icon restrict us to use only Material icons as image?
The Chip should still support images from arbitrary sources (e.g. contact photos)

**Comment by magnus-gross** [comment]
@oKcerG 
I didn't find the original delete button in the icons, maybe I missed it?

Yes, images can be specified and will be shown as a circular image.

I don't know about the letters, is that how Google does it on Android?

**Comment by magnus-gross** [comment]
@plfiorini I don't know if you missed the "Fixes #59" in the commit message, but sure I can change it to "Closes #59" when I incorporate the requested changes

File: src/imports/controls/Chip.qml (hunk 1 above)
Comment (plfiorini) [reply src/imports/controls/Chip.qml]: Icon is an Image it just need the right value to the source property, so if the developer wants to put a custon image he should set for example `icon.source: "http://www.host.ext/path/to/image.png"`

**Comment by magnus-gross** [comment]
I use Icon instead of Image now.

However, I noticed that @oKcerG was correct about the circular icon.
We have to force the image into a circular shape.

Any idea what's the best way to achieve that?

In the worst case, we have to write a shader for that, but maybe there is some easier way.

**Comment by plfiorini** [comment]
We have CircleMask (https://github.com/lirios/fluid/blob/develop/src/imports/effects/CircleMask.qml) see how it is used by CircleImage (https://github.com/lirios/fluid/blob/develop/src/imports/controls/CircleImage.qml).

You can try making the Icon not visible and add a CircleMask like this:

```qml
    CircleMask {
        anchors.fill: iconItem
        source: iconItem
}
```

**Comment by magnus-gross** [comment]
Alright, this did it!

The final problem, is that I still think that `Icon` does not work with images like `Image does`.

If I use `Image` instead of `Icon`, everything works, but if I pass a custom file url when using `Icon`, it does not show the correct image.

File: src/demo/qml/Pages/Controls/ChipPage.qml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * $END_LICENSE$
+ */
+
+import QtQuick 2.10
+import Fluid.Controls 1.0
```
Comment (plfiorini) [inline src/demo/qml/Pages/Controls/ChipPage.qml]: This new type must be for Fluid.Controls 1.1

File: src/imports/controls/Chip.qml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * $END_LICENSE$
+ */
+
+import QtQuick 2.10
+import QtQuick.Controls.Material 2.2
+import Fluid.Controls 1.0
+import Fluid.Effects 1.0
```
Comment (plfiorini) [inline src/imports/controls/Chip.qml]: Set 1.1 to both Fluid imports

File: src/imports/controls/Chip.qml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+/*!
+  \qmltype Chip
+  \inqmlmodule Fluid.Controls
+  \ingroup fluidcontrols
+
+  \brief Chips represent complex entities in small blocks, such as a contact.
```
Comment (plfiorini) [inline src/imports/controls/Chip.qml]: Put an empty line after \brief and add a `\since 1.1`

File: src/imports/controls/qmldir [hunk 5]
```diff
@@ -17,6 +17,7 @@ BottomSheetList 1.0 BottomSheetList.qml
 BottomSheetGrid 1.0 BottomSheetGrid.qml
 CaptionLabel 1.0 CaptionLabel.qml
 Card 1.0 Card.qml
+Chip 1.0 Chip.qml
```
Comment (plfiorini) [inline src/imports/controls/qmldir]: 1.1

File: src/imports/controls/Chip.qml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    */
+    property string caption: ""
+
+    /*!
+      \qmlproperty alias icon
+
+      The icon to display (if any)
+    */
+    property alias icon: contactIcon
+
```
Comment (plfiorini) [inline src/imports/controls/Chip.qml]: add a `property alias font: captionLabel.font`

**Comment by plfiorini** [comment]
Can you elaborate on what happens with the picture?

**Comment by magnus-gross** [comment]
I implemented the requested changes. Note that the version number bump in Chip.qml breaks its demo page for now, so I have made an extra commit for it - you can squash / cherry pick how you like.

About the picture: If I use Icon the external image isn't displayed at all:
https://i.imgur.com/sJkroiM.png

If I use Image instead of Icon, it works though:
https://i.imgur.com/IliuQoO.png

Image has some other problems though, so I would generally prefer Icon as well.

**Pull request merged**